error.go

```diff
@@ -0,0 +1,51 @@
+package iabtcf
+
+import (
+	"fmt"
+)
+
+var (
+	ErrEmptyString      = fmt.Errorf("consent string is empty")
+	ErrNilBits          = fmt.Errorf("bits is nil")
+	ErrNegativeBitIndex = func(index int) error {
+		return fmt.Errorf("negative bit index: %d", index)
+	}
+	ErrBitIndexHigherThanUpperBound = func(index, upperBound int) error {
+		return fmt.Errorf("bit index is higher than upper bound %d: %d", upperBound, index)
+	}
+	ErrBitLowerThanLowerBound = func(number, lowerBound int) error {
+		return fmt.Errorf("bit #%d is lower than lower bound %d", number, lowerBound)
+	}
+	ErrBitHigherThanUpperBound = func(number, upperBound int) error {
+		return fmt.Errorf("bit #%d is higher than upper bound %d", number, upperBound)
+	}
+	ErrInvalidNbBitsMultiple = func(nbBits, multiple int) error {
+		return fmt.Errorf("number of bits is not multiple of %d: %d", multiple, nbBits)
+	}
+
+	ErrDecodeFailed                 = func(cause error) error { return fmt.Errorf("decode failed: %w", cause) }
```

Comment:
What is the use case for exporting these functions?

I feel like these `fmt.Errorf` calls can simply be written inline where they're needed.

Reply:
It was to avoid duplication 😉 
but yes, fair enough 😉 ( since I do not expect the client to treat returned errors specifically ).
✔️ Done